detectors/node/opentelemetry-resource-detector-aws/src/detectors/AwsEcsDetector.ts

```diff
@@ -38,27 +41,65 @@ import { getEnv } from '@opentelemetry/core';
 export class AwsEcsDetector implements Detector {
   readonly CONTAINER_ID_LENGTH = 64;
   readonly DEFAULT_CGROUP_PATH = '/proc/self/cgroup';
+
   private static readFileAsync = util.promisify(fs.readFile);
 
+  private static async getUrlAsJson(url: string): Promise<any> {
```

Comment:
No need for `async` keyword if you aren't using `await`, it just results in unnecessary code generation

Reply:
Done in b04a99f8